Maker 2.0 Launch - Landing pages

kelbyhawn opened the pull request:
Update/add the following landing pages for the Maker 2.0 launch — all pages are fully responsive:

**Note:**  I'll be out when this gets merged, but the Maker team (Dani, Sam, possibly Dan) will do a QA sweep once all Maker 2.0 PRs are merged into staging to double check links, etc. 


- https://code.org/maker
  - Redesigned and updated
  - [Asana task](https://app.asana.com/0/1204178505869103/1204178512655055/f)
- https://code.org/maker/pick-a-device
  - New page
  - [Asana task](https://app.asana.com/0/1204178505869103/1204178512655059/f)
- https://code.org/maker/microbit
  - New page
  - [Asana task](https://app.asana.com/0/1204178505869103/1204178512655075/f)
- https://code.org/maker/circuitplayground
  - Redesigned and updated
  - Redirected https://code.org/circuitplayground to https://code.org/maker/circuitplayground
  - [Asana task](https://app.asana.com/0/1204178505869103/1204178512655077/f)

**Related PRs:**
- https://github.com/code-dot-org/code-dot-org/pull/50870
- https://github.com/code-dot-org/code-dot-org/pull/50761

----

### code.org/maker
![Screenshot 2023-03-23 at 11-20-33 I'm a Maker!](https://user-images.githubusercontent.com/9256643/227310620-6e5ae27f-8f33-4267-8344-02723b9c1c0d.png)

### code.org/pick-a-device
![localhost code org_3000_maker_pick-a-device](https://user-images.githubusercontent.com/9256643/227310814-46101394-e179-40a3-b752-4c3c0e9cccda.png)

### code.org/maker/microbit
![localhost code org_3000_maker_microbit (1)](https://user-images.githubusercontent.com/9256643/227311046-26fddaef-c4d4-4054-a6ff-8d18cc319115.png)

### code.org/maker/circuitplayground
![localhost code org_3000_maker_circuitplayground](https://user-images.githubusercontent.com/9256643/226722976-87ce5f2c-b8ce-493b-acfb-e63037175f19.png)

kelbyhawn commented on shared/css/design-system-pegasus.scss:
```diff
@@ -16,6 +16,8 @@ $border-radius: 4px;
   font-size: 1.25em;
 }
 
+.h5-lookalike-style { @include h5-lookalike-style}
```
Adding this here for now for use in markup — the plan for Phase II of the rebrand updates will include similar heading styles for all h1-h6 tags.

kelbyhawn commented on pegasus/sites.v3/code.org/public/css/maker-skinny-banner.scss:
```diff
@@ -76,26 +77,7 @@
 
-  // There's a mysterious <p> tag being added around
-  // the right image on https://code.org/circuitplayground.
-  // This aligns it correctly to the bottom of the container.
```
This is no longer needed since I updated the /circuitplayground page to a .haml file.

megcrenshaw commented on pegasus/sites.v3/code.org/public/maker/circuitplayground.haml:
```diff
@@ -0,0 +1,97 @@
+%section.intro
+  .wrapper
+    %h1 The Circuit Playground
+    %p.h5-lookalike-style The Circuit Playground is family of small microcontroller boards with LEDs, buttons, and sensors built in.
```
Is this text correct? I have a hard time parsing it as-is. Does there need to be an additional "a"? `The Circuit Playground is *a* family of small microcontroller boards ...`

megcrenshaw replied:
Apologies if this is just copy-and-pasted from the old file

kelbyhawn replied:
Good catch, ty! 🙏

megcrenshaw commented on pegasus/sites.v3/code.org/public/maker/circuitplayground.haml:
```diff
@@ -0,0 +1,97 @@
+%link{href: '/css/generated/maker-styles.css', rel: 'stylesheet'}
+
+-# Import FontAwesome Pro
+%script{src: "https://kit.fontawesome.com/ea2d9d4413.js", crossorigin: "anonymous"}
```
Wow, do you know if we can do this on Dashboard?

kelbyhawn replied:
If only it were so simple haha! This is sort of a long story: this implementation is pulling from an eternal url which is ok for Pegasus since the icons are all decorative so if it gets blocked and the icons disappear it won't affect the UX at all — but since some schools block external urls it's better to have Font Awesome come from our codebase instead on Dashboard since the icons there are more important/informational. 

Ideally, once Font Awesome gets implemented internally we'll switch over to use that here for Pegasus too, but in the meantime this is an ok workaround.

megcrenshaw replied:
Oh, that's good to know! Your explanation makes a lot of sense –– In a future PR, I might go ahead and add it to get the design style right and add a todo to change it back later.

megcrenshaw commented on pegasus/sites.v3/code.org/public/maker/circuitplayground.haml:
```diff
@@ -0,0 +1,97 @@
+    %ul.icon-bullet-list{style: "margin-top: 0"}
+      %li
+        %div
+          %i{class: "#{icon_circle_check}", style: "color: #3EA33E"}
```
This is color-coding weird in the Github browser, making me think there's a syntax error, but I don't see any errors here ... commenting and will come back to this

megcrenshaw replied:
Okay, it's repeatedly happening with the `#` symbol –– not worried about it!

megcrenshaw commented on pegasus/sites.v3/code.org/public/maker/circuitplayground.haml:
```diff
@@ -0,0 +1,97 @@
+          %i{class: "#{icon_circle_xmark}", style: "color: #E5311A"}
+        .text-wrapper.no-margin-bottom
+          %h3 The Circuit Playground Bluefruit
+    %p Wondering which version you have? <a href="https://learn.adafruit.com/introducing-circuit-playground/classic-vs-express" target="_blank" rel="noopener noreferrer">Check out Adafruit's guide</a>.
```
This link doesn't have the Bluefruit version –– does it matter?

kelbyhawn replied:
@dmcavoy can you take a look at this?

dmcavoy replied:
We could go to https://learn.adafruit.com/category/circuit-playground instead which has links to overviews of each board. I can't find a guide with all three. @dancodedotorg fyi

kelbyhawn replied:
Thanks @dmcavoy! I'll update this link to https://learn.adafruit.com/category/circuit-playground with the other minor updates we were talking about in Slack 👍

dancodedotorg replied:
Oh yah - Dani's link is way better. LGTM!

megcrenshaw commented on pegasus/sites.v3/code.org/public/maker/circuitplayground.haml:
```diff
@@ -0,0 +1,97 @@
+      .action-block.action-block--one-col
+        %h3 Circuit Playground Classroom Kit
+        %img{src: "/images/maker/maker-cp-kit.png", alt: "", style: "width: 100%"}
+        %p Contains Circuit Playground boards and accessories necessary for using Code.org's curriculum with a class or 30 students.
```
typo: with a class *of* 30 students

kelbyhawn replied:
Ty! 🙏

megcrenshaw commented on pegasus/sites.v3/code.org/public/maker/circuitplayground.haml:
```diff
@@ -0,0 +1,97 @@
+      .action-block.action-block--one-col
+        %h3 Circuit Playground Individual Kit
+        %img{src: "/images/maker/maker-cp-individual.png", alt: "", style: "width: 100%"}
+        %p{style: "height: 160px"} For smaller classrooms, we recommend investing in a set of Circuit Playground Individual Kits. The individual kit is designed for a single or pair of students.
```
I know I'm not supposed to be copy-editing but this parses weird because you don't say "a single students" ... should it be "a single student or pair of students"

kelbyhawn replied:
I appreciate your copy-edits 😄 I'll update it to "a single student or pair of students".

megcrenshaw commented on pegasus/sites.v3/code.org/views/maker_curriculum_circuit_playground.haml:
```diff
@@ -0,0 +1,17 @@
+        %iframe{src: "https://www.youtube-nocookie.com/embed/9_Z1hvx22us", title: "Creating Apps with Devices", frameborder: "0", allowfullscreen: "true"}
+    %p Explore the role of physical devices in computing. Using App Lab and Adafruit's Circuit Playground, students develop programs that utilize the same hardware inputs and outputs that you see in the smart devices.
+    %p{style: "height: 50px"} <strong>Duration:</strong> 25 hours
+    %a.link-button{href: explore_unit_link, aria:{label: "Explore the Creating Apps with Devices unit"}} Explore the unit
```
TIL about this syntax for aria label! Looks so bizarre. I'm used to `'aria-label'=>` or `'aria-label':` ... haml ...

megcrenshaw replied:
But also love this use of the label! 🥳  (and the others in this PR)

kelbyhawn replied:
I didn't know those were options! 🙀 😂 I googled how to do it and found this implementation — I assume it's like this so you can add more `aria` attributes in there. 

Should I change it to `"aria-label": "Label text here"` or is it ok to leave it how I have it?

megcrenshaw replied:
Oh that makes a lot of sense! I see it in other places in the code base, so I'm good with it –– we don't have norms yet for haml, so I don't think it makes sense to try to "fix" things without clear direction as to what our norms are.

kelbyhawn replied:
Ok cool, sounds good! 

Side note: I like this version bc the color code formatting making it's easier to see it's an attribute, so maybe that's something to consider for when we norm:
<img width="959" alt="Screenshot 2023-03-22 at 10 49 38 AM" src="https://user-images.githubusercontent.com/9256643/226993602-3d30b291-2eab-4de1-8a2b-df217f77a9c7.png">

megcrenshaw commented on pegasus/sites.v3/code.org/public/maker/index.haml:
```diff
@@ -0,0 +1,115 @@
+  .wrapper
+    %div.col-45{style: "margin-bottom: 1em"}
+      %h2 Maker careers
+      %h3.h5-lookalike-style Meet Jerome Morrison at Meow Wolf
```
I love Meow Wolf! Very cool 😎

megcrenshaw commented on shared/css/design-system-pegasus.scss:
```diff
@@ -9,13 +9,19 @@ $border-radius: 4px;
   outline-offset: 2px;
 }
 
+@mixin hr-style {
```
Can you give context about how this works? I don't see it doing anything

kelbyhawn replied:
Oops! It does nothing, I just forgot to delete it 🙃

megcrenshaw reviewed (approved):
Looks amazing, Kelby! Had some minor questions inline (mostly copy-edit questions), but nothing blocking! Love it 🤩

megcrenshaw commented on pegasus/sites.v3/code.org/public/maker/pick-a-device.haml:
```diff
@@ -0,0 +1,12 @@
+%section
+  .wrapper
+    %h1 Pick a device
+    %p Creating apps with devices works with both the Circuit Playground and the micro:bit. Which device do you want to use with your classroom?
```
Nit: Does the unit name need to be capitalized? `Creating Apps with Devices works ...`

kelbyhawn replied:
Yes, great catch! 🙏

kelbyhawn commented:
> Looks amazing, Kelby! Had some minor questions inline (mostly copy-edit questions), but nothing blocking! Love it 🤩

Yay! Thanks for reviewing so quickly @megcrenshaw, and all typos have been updated 😄 🎉

megcrenshaw commented:
> > Looks amazing, Kelby! Had some minor questions inline (mostly copy-edit questions), but nothing blocking! Love it 🤩
> 
> Yay! Thanks for reviewing so quickly @megcrenshaw, and all typos have been updated 😄 🎉

Looks great!

kelbyhawn commented:
@megcrenshaw @dmcavoy I just committed the misc updates — can y'all take a look: [add misc updates](https://github.com/code-dot-org/code-dot-org/pull/50785/commits/a5b3da0775d9a31c72392e5fc7d5755525be0302)

- Added a photo to the intro section on code.org/maker
- Updated the link button labels and aria label variables on the `maker_devices.haml` partial so we can change them on the /maker and /pick-a-device pages
- Misc copy updates/changes

megcrenshaw commented on pegasus/sites.v3/code.org/views/maker_devices.haml:
```diff
@@ -0,0 +1,13 @@
+  .action-block.action-block--one-col
+    %h3 Circuit Playground
+    %img{src: "/images/maker/maker-device-circuit-playground.png", alt: ""}
+    %a.link-button{href: curriculum_link_cp, aria:{label: aria_label_cp}}
```
Where is `aria_label_cp` defined? I see it elsewhere in the repo, but not sure how it's getting through to here.

megcrenshaw replied:
Oh ignore me! Just looked more closely about how it's getting passed in –– looks great!

megcrenshaw reviewed (approved):
Updates look great!

kelbyhawn commented:
> Updates look great!

Awesome, thank you! 🙌
